### ISSUE TITLE
Select All functionality - performance issue for 50K+ values (Unresponsive page)

### ISSUE THREAD

**Issue by sasivardhan518** [ng-select/ng-select#1827]
**Describe the bug**
Select All functionality - performance issue (Unresponsive page),
even when virtual scroll is enabled and when we have 50k+ values

**Reproducible example**
https://stackblitz.com/edit/angular-lxlzxq-jz4eqm?file=src%2Ftemplate-header-footer-example.component.ts

**To Reproduce** 
1. Open dropdown and click on Select All.
2. Observe the page becomes unresponsive and popup gets displayed.

**Additional context**
when i tried debugging the issue, found the problem to be within the below function from the ng-select code **(ng-select.component.ts)**. Depending upon how many items we are feeding to ng-select it is running n*(n(n+1)/2) heavy looping which we should ideally avoid by using some kind of map object.

 ```
private _handleWriteValue(ngModel: any | any[]) {
        if (!this._isValidWriteValue(ngModel)) {
            return
        }

        const select = (val: any) => {
            let item = this.itemsList.findItem(val);
            if (item) {
                this.itemsList.select(item);
            } else {
                const isValObject = isObject(val);
                const isPrimitive = !isValObject && !this.bindValue;
                if ((isValObject || isPrimitive)) {
                    this.itemsList.select(this.itemsList.mapItem(val, null));
                } else if (this.bindValue) {
                    item = {
                        [this.bindLabel]: null,
                        [this.bindValue]: val
                    };
                    this.itemsList.select(this.itemsList.mapItem(item, null));
                }
            }
        };

        if (this.multiple) {
            (<any[]>ngModel).forEach(item => select(item));
        } else {
            select(ngModel);
        }
    }
```

**Comment by bhaskerchari** [comment]
We have also observed similar performance issues in our application where we have enabled "Select All"/"Clear All" behavior for all usages of ng-select and when the values are huge the app is running in to page unresponsive error and browser just freezes.

I think this issue can only be addressed from within the library by leveraging maps instead of iterating through huge arrays and finding a match to select an item and can't do much anything as a work around unless we get rid of  providing "Select All" option which is limiting the usability and user flexibility.

Hope someone would give a quick shot at it as it seems to be the quite common usage and may easily run into this usecase.
